sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/data/MutableDoublePointData.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.data;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.metrics.data.DoubleExemplarData;
+import io.opentelemetry.sdk.metrics.data.DoublePointData;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A mutable {@link DoublePointData}
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ *
+ * <p>This class is not thread-safe.
+ */
+public class MutableDoublePointData implements DoublePointData {
+
+  private long startEpochNanos;
+  private long epochNanos;
+
+  private Attributes attributes = Attributes.empty();
+
+  private double value;
+  private List<DoubleExemplarData> exemplars = Collections.emptyList();
+
+  @Override
+  public double getValue() {
+    return value;
+  }
+
+  @Override
+  public long getStartEpochNanos() {
+    return startEpochNanos;
+  }
+
+  @Override
+  public long getEpochNanos() {
+    return epochNanos;
+  }
+
+  @Override
+  public Attributes getAttributes() {
+    return attributes;
+  }
+
+  @Override
+  public List<DoubleExemplarData> getExemplars() {
+    return exemplars;
+  }
+
+  /**
+   * Sets all {@link MutableDoublePointData} values based on {@code point}.
+   *
+   * @param point The point to take the values from
+   */
+  public void set(DoublePointData point) {
+    set(
+        point.getStartEpochNanos(),
+        point.getEpochNanos(),
+        point.getAttributes(),
+        point.getValue(),
+        point.getExemplars());
+  }
+
+  /** Sets all {@link MutableDoublePointData} values , besides exemplars which are set to empty. */
+  public void set(long startEpochNanos, long epochNanos, Attributes attributes, double value) {
+    set(startEpochNanos, epochNanos, attributes, value, Collections.emptyList());
+  }
```

Comment:
#nit: this overload obfuscates what happens with exemplars. Better the caller explicitly set exemplars to `Collections.emptyList()` IMO.

Reply:
I modeled it by existing
```
  public static DoublePointData create(
      long startEpochNanos, long epochNanos, Attributes attributes, double value) {
    return create(startEpochNanos, epochNanos, attributes, value, Collections.emptyList());
```

I can change both if you want?